frontend/src/containers/Table.js

```diff
@@ -0,0 +1,268 @@
+    const { windowId, docId, updateDocList, tabId } = this.props;
+
+    this.handleSelect();
+
+    // TODO: This should be an action creator
+    deleteRequest(
+      'window',
+      windowId,
+      docId ? docId : null,
+      docId ? tabId : null,
+      selected
+    ).then(() => {
+      if (!docId) {
+        updateDocList();
+      }
+    });
+  };
+
+  handleZoomInto = (fieldName) => {
+    const { entity, windowId, docId, tabId, viewId, selected } = this.props;
+
+    getZoomIntoWindow(
+      entity,
+      windowId,
+      docId,
+      entity === 'window' ? tabId : viewId,
+      selected[0],
+      fieldName
+    ).then((res) => {
+      res &&
+        res.data &&
+        window.open(
+          `/window/${res.data.documentPath.windowId}/${
+            res.data.documentPath.documentId
+          }`,
+          '_blank'
+        );
+    });
+  };
+
+  handleRowCollapse = (node, collapse) => {
+    const { collapseTableRow, windowId, viewId, tabId, docId } = this.props;
+    const tableId = getTableId({ windowId, viewId, docId, tabId });
+
+    collapseTableRow({ tableId, collapse, node });
+  };
+
+  render() {
+    return (
+      <Table
+        {...this.props}
+        onHandleZoomInto={this.handleZoomInto}
+        onPromptSubmit={this.handlePromptSubmit}
+        onItemChange={this.handleItemChange}
+        onSelect={this.handleSelect}
+        onSelectAll={this.handleSelectAll}
+        onDeselectAll={this.handleDeselectAll}
+        onDeselect={this.handleDeselect}
+        onRowCollapse={this.handleRowCollapse}
+        onGetAllLeaves={this.getAllLeaves}
+        onHandleAdvancedEdit={this.handleAdvancedEdit}
+        onOpenTableModal={this.openTableModal}
+      />
+    );
+  }
+}
+
+TableContainer.propTypes = containerPropTypes;
+
+const mapStateToProps = (state, props) => {
+  const { windowId, docId, tabId, viewId } = props;
+  const tableId = getTableId({ windowId, viewId, docId, tabId });
+  const table = getTable(state, tableId);
+  const modalVisible = state.windowHandler.modal.visible;
+  let handleShortcuts = state.windowHandler.allowShortcut;
+
+  // we don't have to worry about shortcuts if table is behind a modal
+  if (modalVisible && !props.isModal) {
+    handleShortcuts = false;
+  }
+
+  return {
+    rows: table.rows,
+    columns: table.columns,
+    selected: table.selected,
+    pending: table.dataPending,
+    collapsedParentRows: table.collapsedParentRows,
+    collapsedRows: table.collapsedRows,
+    collapsedArrayMap: table.collapsedArrayMap,
+    activeSort: table.activeSort,
+    emptyText: table.emptyText,
+    emptyHint: table.emptyHint,
+    indentSupported: table.indentSupported,
+    collapsible: table.collapsible,
+    keyProperty: table.keyProperty,
+    size: table.size,
+    allowShortcut: handleShortcuts,
+    allowOutsideClick: state.windowHandler.allowOutsideClick,
+    modalVisible,
+    isGerman:
+      state.appHandler.me.language && state.appHandler.me.language.key
```

Comment:
This is not quite reliable to get the language. I found out that the most reliable way is from the localStore where it is set as a result of the call to getSession.  See ex: https://github.com/metasfresh/metasfresh/pull/6824/files#diff-858b40143f05e165f6bc793cd0a6fa32R303

Reply:
This is not in `BIG_BANG` yet, so I can update that after I'll rebase it with master.